planning/intersection_types/implementation_plan.md

```
@@ -56,87 +56,90 @@ This plan outlines the work needed to fully support intersection types with Type
-	// Step 3: If still an intersection, check for mergeable object types
-	if inter, ok := result.(*IntersectionType); ok {
-		// Check if all types are object types that can be merged
-		allObjects := true
-		for _, typ := range inter.Types {
-			if _, ok := typ.(*ObjectType); !ok {
-				allObjects = false
-				break
-			}
-		}
+func (c *Checker) NormalizeIntersectionType(ctx Context, t *type_system.IntersectionType) type_system.Type {
+	// Step 1: Prune and expand all types to resolve type variables and type aliases
```

markdownlint-cli2 (0.18.1)

Hard tabs
Column: 1

(MD010, no-hard-tabs)